collector/collector.go

```diff
@@ -17,36 +17,45 @@
 package collector
 
 import (
+	"fmt"
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 	"github.com/prometheus/client_golang/prometheus"
+	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var (
+	collectorLog = ctrl.Log.WithName("collector")
+)
+
 // PipelineServiceCollector struct
 type PipelineServiceCollector struct {
 	logger            log.Logger
-	durationScheduled *prometheus.GaugeVec
-	durationCompleted *prometheus.GaugeVec
+	durationScheduled *prometheus.HistogramVec
+	durationCompleted *prometheus.HistogramVec
 
 	client client.Client
 }
 
 func NewCollector(logger log.Logger, client client.Client) (*PipelineServiceCollector, error) {
 	//TODO should this be converted to a Desc so we can use constant metrics
-	durationScheduled := prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "pipelinerun_duration_scheduled_seconds",
-		Help: "Duration in seconds for a PipelineRun to be scheduled.",
-		//TODO the uid will get us a unique entry, but wouldn't adding namespace will help usability when consuming this metric?
-		// also, labels at an individual PipelineRun is a scalability issue with prometheus
-	}, []string{"name", "uid"})
+	// shipwright establishment metrics buckets is []float64{0, 1, 2, 3, 5, 7, 10, 15, 20, 30}
+	// shipwright ramp up metrics buckets is prometheus.LinearBuckets(0, 1, 10)
+	durationScheduled := prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "pipelinerun_duration_scheduled_seconds",
+		Help:    "Duration in seconds for a PipelineRun to be scheduled.",
+		Buckets: prometheus.LinearBuckets(0, 1, 10),
+	}, []string{"namespace"})
 
 	//TODO should this be converted to a Desc so we can use constant metrics
-	durationCompleted := prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "pipelinerun_duration_completed_seconds",
-		Help: "Duration in seconds for a PipelineRun to complete.",
-		//TODO the uid will get us a unique entry, but wouldn't adding namespace help usability with humans consuming this metric?
-	}, []string{"name", "uid"})
+	// shipwright completion duration buckets is prometheus.LinearBuckets(50, 50, 10)
```

Comment:
will remove before we merge - just wanted to provide some context on the bucket settings from other projects that I took as a starting point